server/server.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2019 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/0xRLG/ocworkqueue"
+	"go.opencensus.io/stats/view"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/workqueue"
+
+	"metacontroller.app/apis/metacontroller/v1alpha1"
+	mcclientset "metacontroller.app/client/generated/clientset/internalclientset"
+	mcinformers "metacontroller.app/client/generated/informer/externalversions"
+	"metacontroller.app/controller/composite"
+	"metacontroller.app/controller/decorator"
+	dynamicclientset "metacontroller.app/dynamic/clientset"
+	dynamicdiscovery "metacontroller.app/dynamic/discovery"
+	dynamicinformer "metacontroller.app/dynamic/informer"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+)
+
+type controller interface {
+	Start()
+	Stop()
+}
+
+func Start(config *rest.Config, discoveryInterval, informerRelist time.Duration) (stop func(), err error) {
+	// Periodically refresh discovery to pick up newly-installed resources.
+	dc := discovery.NewDiscoveryClientForConfigOrDie(config)
+	resources := dynamicdiscovery.NewResourceMap(dc)
+	// We don't care about stopping this cleanly since it has no external effects.
+	resources.Start(discoveryInterval)
+
+	// Create informerfactory for metacontroller api objects.
```

Comment:
nit: `informer factory`, so to be coherent with others such as `dynamic clientset` for `dynamicclientset`.